File: app/src/main/java/edu/ksu/wheatgenetics/verify/SettingsActivity.java
package edu.ksu.wheatgenetics.verify;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;

public class SettingsActivity extends AppCompatActivity {

    public static String SCAN_MODE_LIST = "edu.ksu.wheatgenetics.verify.SCAN_MODE";
    public static String AUDIO_ENABLED = "edu.ksu.wheatgenetics.verify.AUDIO_ENABLED";
    public static String TUTORIAL_MODE = "edu.ksu.wheatgenetics.verify.TUTORIAL_MODE";
    public static String USER_NAME = "edu.ksu.wheatgenetics.verify.USER_NAME";
    public static String LIST_KEY_NAME = "edu.ksu.wheatgenetics.verify.LIST_KEY_NAME";

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if(getSupportActionBar() != null){
            getSupportActionBar().setTitle(null);
            getSupportActionBar().getThemedContext();
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setHomeButtonEnabled(true);
        }

        getFragmentManager().beginTransaction()
                .replace(android.R.id.content, new SettingsFragment())
                .commit();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        switch (item.getItemId()) {
            case android.R.id.home:
                setResult(RESULT_OK);
                finish();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/java/edu/ksu/wheatgenetics/survey/SettingsActivity.kt
package edu.ksu.wheatgenetics.survey

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity

/**
 * Created by Chaney on 4/20/2017.
 */

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (getSupportActionBar() != null) {
            getSupportActionBar()?.setTitle(null)
            getSupportActionBar()?.getThemedContext()
            getSupportActionBar()?.setDisplayHomeAsUpEnabled(true)
            getSupportActionBar()?.setHomeButtonEnabled(true)
        }

        fragmentManager.beginTransaction()
                .replace(android.R.id.content, SettingsFragment())
                .commit()
    }


    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        when (item.itemId) {
            android.R.id.home -> {
                setResult(RESULT_OK)
                finish()
                return true
            }
            else -> return super.onOptionsItemSelected(item)
        }
    }

    companion object {

        var PERSON = "edu.ksu.wheatgenetics.survey.PERSON"
        var EXPERIMENT = "edu.ksu.wheatgenetics.survey.EXPERIMENT_ID"
        var MIN_ACCURACY = "edu.ksu.wheatgenetics.survey.MIN_ACCURACY"
        var MIN_DISTANCE = "edu.ksu.wheatgenetics.survey.MIN_DISTANCE"
        var MIN_TIME = "edu.ksu.wheatgenetics.survey.MIN_TIME"
    }
}